drivers/modem/ublox-sara-r4.c

```diff
@@ -1304,85 +660,30 @@ static void modem_reset(void)
 		goto error;
 	}
 
-	/* echo on */
-	ret = send_at_cmd(NULL, "ATE1", MDM_CMD_TIMEOUT);
-	if (ret < 0) {
-		LOG_ERR("ATE1 ret:%d", ret);
-		goto error;
-	}
-
-	/* stop functionality */
-	ret = send_at_cmd(NULL, "AT+CFUN=0", MDM_CMD_TIMEOUT);
-	if (ret < 0) {
-		LOG_ERR("AT+CFUN=0 ret:%d", ret);
-		goto error;
-	}
-
-#if defined(CONFIG_BOARD_PARTICLE_BORON)
-	/* use external SIM */
-	ret = send_at_cmd(NULL, "AT+UGPIOC=23,0,0", MDM_CMD_TIMEOUT);
-	if (ret < 0) {
-		LOG_ERR("AT+UGPIOC=23,0,0 ret:%d", ret);
-		goto error;
-	}
-
-	/* Let SIM settle */
-	k_sleep(MDM_CMD_TIMEOUT);
-#endif
-
-	/* UNC messages for registration */
-	ret = send_at_cmd(NULL, "AT+CREG=1", MDM_CMD_TIMEOUT);
-	if (ret < 0) {
-		LOG_ERR("AT+CREG=1 ret:%d", ret);
-		goto error;
-	}
-
-	/* HEX receive data mode */
-	ret = send_at_cmd(NULL, "AT+UDCONF=1,1", MDM_CMD_TIMEOUT);
-	if (ret < 0) {
-		LOG_ERR("AT+UDCONF=1 ret:%d", ret);
-	}
-
-	/* query modem info */
-	LOG_INF("Querying modem information");
-	ret = send_at_cmd(NULL, "ATI", MDM_CMD_TIMEOUT);
-	if (ret < 0) {
-		LOG_ERR("ATI ret:%d", ret);
-		goto error;
-	}
-
-	/* query modem IMEI */
-	ret = send_at_cmd(NULL, "AT+CGSN", MDM_CMD_TIMEOUT);
-	if (ret < 0) {
-		LOG_ERR("AT+CGSN ret:%d", ret);
-		goto error;
-	}
-
-	/* setup PDP context definition */
-
-	ret = send_at_cmd(NULL, "AT+CGDCONT=1,\"IP\",\""
-				CONFIG_MODEM_UBLOX_SARA_R4_APN "\"",
-			  MDM_CMD_TIMEOUT);
+	ret = modem_cmd_handler_setup_cmds(&mctx.iface, &mctx.cmd_handler,
```

Comment:
The function name contains a bit redundant data, would `modem_cmd_handler_setup()` be descriptive enough?

Reply:
Naming is hard :)  I'm not sure `modem_cmd_handler_setup()` is descriptive enough as that sounds like you're setting up a cmd_handler.

This function runs a bunch of setup commands via the command handler..  Not sure how to make that name better.

Reply:
No worries, there was just many cmd chars in the function name. We can leave it like this, it is not really an issue.